disnake/state.py

```diff
@@ -777,12 +778,19 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
         # we ensure that the channel is a type that implements last_message_id
         if channel and channel.__class__ in (TextChannel, Thread, VoiceChannel):
             channel.last_message_id = message.id  # type: ignore
+        if channel and isinstance(channel, Thread):
+            channel.total_message_sent += 1
+            channel.message_count += 1
 
     def parse_message_delete(self, data: gateway.MessageDeleteEvent) -> None:
         raw = RawMessageDeleteEvent(data)
         found = self._get_message(raw.message_id)
         raw.cached_message = found
         self.dispatch("raw_message_delete", raw)
+        guild = self._get_guild(raw.guild_id)
+        thread = guild and guild.get_thread(raw.channel_id)
+        if thread:
+            thread.message_count -= 1
```

Comment:
it is possible for this to be None (and this goes for all of the other uses in this file) which would cause a TypeError when trying to subtract one, but please wait for resolving if this should actually be optional, per a different comment concerning the typings.

Additionally, this may have issues with threads older than july and result in a negative message_count, although it may not be worth catching that rare case...

Reply:
For the second part of your comment:

maybe we can check if the `created_at` property of the thread is older than July 2, 2022 23:59:59, if it is we ignore it, what do you think? maybe there's other ways

Reply:
> this may have issues with threads older than july and result in a negative message_count

`thread.message_count = max(0, thread.message_count - 1)` would avoid that